### PR TITLE
Procedures that wrap AWS NLP Entities endpoint

### PR DESCRIPTION
**Pull request by mneedham**
Just to start with. We'll wrap the others as well in upcoming PRs.

### CODE REVIEW

File: src/main/kotlin/apoc/nlp/aws/AWSProcedures.kt [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+        private fun convert(source: Any): List<Node> {
+            return when (source) {
+                is Node -> listOf(source)
+                is List<*> -> source.map { item -> item as Node }
+                else -> throw java.lang.IllegalArgumentException("`source` must be a node or list of nodes, but was: `${source}`")
+            }
+        }
+
+        private fun verifySource(source: Any) {
```
Comment (conker84) [inline src/main/kotlin/apoc/nlp/aws/AWSProcedures.kt:86]: can we move this and the other validation methods to Util class and reuse them in AWS and GCP procedures?

File: src/test/kotlin/apoc/nlp/aws/AWSProceduresAPITest.kt [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+
+        fun nodeMatches(item: Node?, labels: List<String>?, properties: Map<String, Any>?): Boolean {
+            val labelsMatched = item?.labels!!.count() == labels?.size  && item.labels!!.all { label -> labels?.contains(label.name())!! }
+            val propertiesMatches = item.allProperties.keys == properties?.keys && item.allProperties.all { entry -> properties?.containsKey(entry.key)!! && properties.get(entry.key) == entry.value }
+            return labelsMatched && propertiesMatches
+        }
+    }
+
+    @Test
+    fun `should extract entities for individual nodes`() {
```
Comment (conker84) [inline src/test/kotlin/apoc/nlp/aws/AWSProceduresAPITest.kt:57]: there is no assertion here, is it correct?

File: src/test/kotlin/apoc/nlp/aws/AWSProceduresAPITest.kt [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+                    })
+                    YIELD value
+                    RETURN value
+                """.trimIndent(), mapOf("apiKey" to apiKey, "apiSecret" to apiSecret)) {
+            println(it.resultAsString())
+        }
+    }
+
+    @Test
+    fun `should extract entities for collection of nodes`() {
```
Comment (conker84) [inline src/test/kotlin/apoc/nlp/aws/AWSProceduresAPITest.kt:78]: there is no assertion here, is it correct?

File: src/test/kotlin/apoc/nlp/aws/NodeMatcher.kt [hunk 4]
```diff
@@ -0,0 +1,25 @@
+package apoc.nlp.aws
+
+import org.hamcrest.Description
+import org.neo4j.graphdb.Label
+import org.neo4j.graphdb.Node
+import java.util.stream.Collectors
+
+class NodeMatcher(labels: List<Label>, val properties: Map<String, Any>) : org.hamcrest.TypeSafeDiagnosingMatcher<Node>() {
```
Comment (conker84) [inline src/test/kotlin/apoc/nlp/aws/NodeMatcher.kt:8]: can we use a data class?

File: src/test/kotlin/apoc/nlp/aws/RelationshipMatcher.kt [hunk 5]
```diff
@@ -0,0 +1,35 @@
+package apoc.nlp.aws
+
+import org.hamcrest.Description
+import org.neo4j.graphdb.Node
+import org.neo4j.graphdb.Relationship
+
+class RelationshipMatcher(private val startNode: Node?, private val endNode: Node?, private val relationshipType: String) : org.hamcrest.TypeSafeDiagnosingMatcher<Relationship>() {
```
Comment (conker84) [inline src/test/kotlin/apoc/nlp/aws/RelationshipMatcher.kt:7]: can we use a data class?

File: src/main/kotlin/apoc/nlp/NLPHelperFunctions.kt [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+import org.neo4j.graphdb.Node
+import org.neo4j.graphdb.Relationship
+import org.neo4j.graphdb.RelationshipType
+import org.neo4j.graphdb.Transaction
+import java.util.stream.Stream
+
+class NLPHelperFunctions {
+    companion object {
+        fun createRelationships(node: Node, nodes: MutableSet<Node>, relationshipType: RelationshipType) =
+                sequence {
```
Comment (conker84) [inline src/main/kotlin/apoc/nlp/NLPHelperFunctions.kt:13]: do we need this lazy evaluation here? or we can just do a `map` over `nodes`?

File: docs/asciidoc/nlp/aws.adoc [hunk 7]
```diff
@@ -0,0 +1,273 @@
...
+==== Install AWS Java SDK For Amazon Comprehend
+
+The AWS NLP procedures are compiled against version _1.11.163_ of the https://mvnrepository.com/artifact/com.amazonaws/aws-java-sdk-comprehend[AWS Java SDK For Amazon Comprehend^] library, which is not included in the APOC Library.
+To use the library we'll need to download the following jar files and place them in the `plugins` directory:
+
+* https://repo1.maven.org/maven2/com/amazonaws/aws-java-sdk-comprehend/1.11.683/aws-java-sdk-comprehend-1.11.683.jar[aws-java-sdk-comprehend-1.11.683.jar^]
+* https://repo1.maven.org/maven2/com/amazonaws/aws-java-sdk-core/1.11.683/aws-java-sdk-core-1.11.683.jar[aws-java-sdk-core-1.11.683.jar^]
+* https://repo1.maven.org/maven2/org/apache/httpcomponents/httpclient/4.5.9/httpclient-4.5.9.jar[httpclient-4.5.9.jar^]
+* https://repo1.maven.org/maven2/org/apache/httpcomponents/httpcore/4.4.11/httpcore-4.4.11.jar[httpcore-4.4.11.jar^]
+* https://repo1.maven.org/maven2/joda-time/joda-time/2.8.1/joda-time-2.8.1.jar[joda-time-2.8.1.jar^]
```
[the remark this answers is not in the document]
Comment (mneedham) [reply docs/asciidoc/nlp/aws.adoc:46]: Can you share the build config that you used to do that? We can add it to the build and then publish it to the GH release page

File: src/main/kotlin/apoc/nlp/NLPHelperFunctions.kt (hunk 6 above)
Comment (mneedham) [reply src/main/kotlin/apoc/nlp/NLPHelperFunctions.kt:13]: oh yeh good point, dunno why I complicated it

**Pull request merged**